src/wagtail_vector_index/index/model.py

```diff
@@ -38,7 +38,9 @@ def get_documents(self) -> Iterable[Document]:
         querysets = self._get_querysets()
 
         for queryset in querysets:
-            for instance in queryset.prefetch_related("embeddings"):
+            for instance in queryset.prefetch_related(
+                "embeddings", "embeddings__content_type"
```

Comment:
Given we're only getting the content type id below, why not use `content_type_id": embedding.content_type_id,`, thus avoiding the need for the prefetch?

Reply:
We control the embedding model so `content_type_id` is the same as `pk`, but we should never assume that the related model's foreign key field has values from the `pk` field. I'm always in two minds about this.

Reply:
> we should never assume that the related model's foreign key field has values from the `pk` field

I don't follow. We define `content_type` as a `ForeignKey`, so it will point to the model's `pk` (however it may be defined), and accessible as `content_type_id` (regardless of the related model's naming of its primary key field). So it seems to me we should simply use `content_type_id` as @zerolab suggests.

Reply:
Agree. It was more of a philosophical point and seeing projects switch their PKs down the line was a maintenance hell. In this case, it's fine to use `content_type_id`.